Add creation of dynamic voice channels

nbrady-techempower opened the pull request:
### Problem to Solve
 
 Right now, joining voice channels related to a game is a little sloppy. Players have to claim a channel and make sure that only players on their team are inside.
 
 ### Solution
 
 After players have checked in to a game, the bot creates 3 voice channels using the game id. One channel is the public channel, and then a red team channel that only players on the red team can see and join, and a blue team channel that only players on the blue team can see and join.
 
 When the game is scored (or canceled) the bot cleans up these channels.

### Notes

The bot will need the appropriate permissions for managing voice channels. 

Testing for this seems difficult the way testing is currently set up (and with my fresh knowledge of the Discord API), so I did it manually to confirm it worked.

mrtolkien commented:
Sounds super interesting!

It will need a few cleanups before merging though:
- Needs to be backwards compatible with the current version, which means by default the bot should *not* create voice channels
- Needs to be managed by admins (most likely a new command in `AdminCog`)
- Needs dynamic options for naming the categories and such (`VOICE_CATEGORY = '▬▬ Team Voice Chat ▬▬'`, ...), could likely just be through environment variables, and defaults to the current values you gave it

nbrady-techempower commented:
@mrtolkien That all makes sense. I'll work on that.

nbrady-techempower commented:
@mrtolkien I've addressed the concerns here. I've created a `server_config` table that holds a JSON config. I've set this up in such a way that new keys can be added in the constants.py without adding any type of migrations. Initially, when the app tries to fetch the config for the first time, it will write a config to the db if it doesn't already exist. After that, all new keys will return `False` until they're turned on.

To make it so there wasn't a mess of new commands for every feature, I've added an `!admin config <key> <option>` command that looks like this:

![image](https://user-images.githubusercontent.com/1304934/110877001-7b908300-828d-11eb-9452-6917df1c7367.png)


I've also set it up, currently, such that the feature handlers will be the ones that check whether the feature is enabled or disabled, instead of worrying about that conditional check everywhere you call a feature function. This is just personal preference so you can let me know if you'd like that changed.

We talked briefly about extracting game start / cleanup steps. I can look at that while working on the next feature.

mrtolkien reviewed (approved):
Everything looks good!

victorcstv commented:
I tried to use these commands but that not working here, take a look...

![image](https://user-images.githubusercontent.com/62709537/145704258-309f6a0c-9ea1-41ba-abfa-66483f0f4bff.png)

mrtolkien commented:
This was never released and neither Discord.py nor this project are maintained anymore.
